python/ray/tune/trial.py

```diff
@@ -162,15 +219,21 @@ def __init__(self,
 
         self.export_formats = export_formats
         self.status = Trial.PENDING
-        self.logdir = None
         self.runner = None
-        self.result_logger = None
         self.last_debug = 0
         self.error_file = None
         self.error_msg = None
         self.num_failures = 0
         self.custom_trial_name = None
 
+        self.trial_dir_schema = TrialDirSchema(str(self), self.local_dir)
+        self.logdir = self.trial_dir_schema.logdir
+        self.trial_dir_schema.mkdir()
+        self.syncer = get_syncer(self.trial_dir, self.trial_dir,
```

Comment:
Why not create this when the result_logger is created? 

Can you also rename this to `checkpoint_syncer`? Otherwise, it's hard to immediately know the difference.

Reply:
keeping as syncer since it syncs results too